bin/plugin/commands/changelog.js

```diff
@@ -757,6 +775,94 @@ function sortFeatureGroups( featureGroups ) {
 	);
 }
 
+/**
+ *
+ * @param {IssuesListForRepoResponseItem[]} pullRequests List of pull requests.
+ *
+ * @return {IssuesListForRepoResponseItem[]} pullRequests List of first time contributor PRs.
+ */
+function getFirstTimeContributorPRs( pullRequests ) {
+	return pullRequests.filter( ( pr ) => {
+		if ( pr.user.login === 'dependabot[bot]' ) {
+			return false;
+		}
+
+		return pr.labels.find(
+			( { name } ) => name.toLowerCase() === 'first-time contributor'
+		);
+	} );
+}
+
+/**
+ *
+ * @param {IssuesListForRepoResponseItem[]} ftcPRs List of first time contributor PRs.
+ *
+ * @return {string} The formatted markdown list of contributors and their PRs.
+ */
+function getContributorMarkdownList( ftcPRs ) {
+	return ftcPRs.reduce( ( markdownList, pr ) => {
+		const title = getNormalizedTitle( pr.title, pr ) || '';
+
+		const formattedTitle = getFormattedItemDescription(
+			title,
+			pr.number,
+			pr.pull_request.html_url
+		);
+
+		markdownList +=
+			'- ' + '@' + pr.user.login + ': ' + formattedTitle + '\n';
+		return markdownList;
+	}, '' );
+}
+
+/**
+ *
+ * @param {IssuesListForRepoResponseItem[]} pullRequests List of pull requests.
+ *
+ * @return {string} The formatted changelog string.
+ */
+function formatContributors( pullRequests ) {
+	const contributorsList = flow( [
+		getFirstTimeContributorPRs,
+		getContributorMarkdownList,
+	] )( pullRequests );
+
+	return (
+		'## First time contributors' +
+		'\n\n' +
+		'The following PRs were merged by first time contrbutors:' +
+		'\n\n' +
+		contributorsList
+	);
+}
+
+/**
+ * @param {WPChangelogSettings} settings Changelog settings.
+ *
+ * @return {Promise<string>} Promise resolving to contributors list.
+ */
+async function getContributors( settings ) {
+	const octokit = new Octokit( {
+		auth: settings.token,
+	} );
+
+	const pullRequests = await fetchAllPullRequests( octokit, settings );
```

Comment:
It looks like we call the same request twice:
- for changelo
- for new contributors 

How about it gets lifted outside and passed to both functions?

Reply:
Yeh I was wondering about that. Ok I'll extract it 👍

Reply:
@gziolo I've completed it and in the process possibly uncovered a bug. I was previously resolving two promises and using the `PromiseResult` as the output to be `log`'d.

Now by extracting I've removed the need to have so many async functions and also made the concat'ing of the two formatted strings a lot clearer.

I would appreciate another quick 👀 from you on this as it's changed a fair amount since your original 👍 .